Proposal.md

```diff
@@ -27,86 +28,93 @@ This proposal adds a `RangeSet` type for representing multiple, noncontiguous ra
 
 // Reset `numbers`
 numbers = Array(1...15)
 
 // You can also build range sets by hand using array literals...
-let myRangeSet: RangeSet = [0..<5, 10..<15]
-print(Array(numbers[myRangeSet]))
+let notTheMiddle: RangeSet = [0..<5, 10..<15]
+print(Array(numbers[notTheMiddle]))
 // Prints [1, 2, 3, 4, 5, 11, 12, 13, 14, 15]
 
 // ...or by using set operations
-let evenThrees = indicesOfThree.intersection(
-    numbers.indices(where: { $0.isMultiple(of: 2) }))
-print(Array(numbers[evenThrees]))
-// Prints [6, 12]
+let smallEvens = indicesOfEvens.intersection(
+    numbers.indices(where: { $0 < 10 }))
+print(Array(numbers[smallEvens]))
+// Prints [2, 4, 6, 8]
 ```
 
-The remainder of the `RangeSet` and collection operations, like inverting a range set or inserting and removing range expressions, are covered in the next section.
+These are just a few examples; the complete proposal includes operations like inverting a range set and inserting and removing ranges, which are covered in the next section.
 
 ## Detailed design
 
-The `RangeSet` type is generic over any `Comparable` type, and supports fast containment checks for individual values, as well as adding and removing ranges of that type. 
+`RangeSet` is generic over any `Comparable` type, and supports fast containment checks for ranges and individual values, as well as adding and removing ranges of that type. 
 
 ```swift
 /// A set of ranges of any comparable value.
 public struct RangeSet<Bound: Comparable> {
     /// Creates an empty range set.
     public init() {}
 
-    /// Creates a range set with the given range.
-    ///
-    /// - Parameter range: The range to use for the new range set.
+    /// Creates a range set containing the given range.
     public init(_ range: Range<Bound>)
     
-    /// Creates a range set with the given ranges.
-    ///
-    /// - Parameter ranges: The ranges to use for the new range set.
+    /// Creates a range set containing the given ranges.
     public init<S: Sequence>(_ ranges: S) where S.Element == Range<Bound>
     
     /// A Boolean value indicating whether the range set is empty.
     public var isEmpty: Bool { get }
     
-    /// Returns a Boolean value indicating whether the given element is
-    /// contained in the range set.
-    ///
-    /// - Parameter element: The element to look for in the range set.
-    /// - Return: `true` if `element` is contained in the range set; otherwise,
-    ///   `false`.
+    /// Returns a Boolean value indicating whether the given value is
+    /// contained by the ranges in the range set.
     ///
     /// - Complexity: O(log *n*), where *n* is the number of ranges in the
     ///   range set.
-    public func contains(_ element: Bound) -> Bool
+    public func contains(_ value: Bound) -> Bool
         
-    /// Inserts the given range into the range set.
+    /// Returns a Boolean value indicating whether the given range is
+    /// contained by the ranges in the range set.
```

Comment:
I'm not sure you need to say "contained by the ranges."  Haven't seen how you sorted out the "what is it, really?" question yet, but my instinct says, *if* we're going to call this thing a set, it's a potentially-infinitely-large non-enumerable set of ranges, in its minimal representation.  In other words, `0.1 ... 0.2` is just as much an element of `[ 0.0 ... 10.0 ]` as `0.0 ... 10.0` is.

Reply:
Either that, or we can't really use the word "contains" for this operation ;-).